addons/hr_timesheet/controllers/project.py

```diff
@@ -39,14 +39,9 @@ def _task_get_page_view_values(self, task, access_token, **kwargs):
         values = super(ProjectCustomerPortal, self)._task_get_page_view_values(task, access_token, **kwargs)
         domain = request.env['account.analytic.line']._timesheet_get_portal_domain()
         task_domain = expression.AND([domain, [('task_id', '=', task.id)]])
-        subtask_domain = expression.AND([domain, [('task_id', 'in', task.child_ids.ids)]])
         timesheets = request.env['account.analytic.line'].sudo().search(task_domain)
-        subtasks_timesheets = request.env['account.analytic.line'].sudo().search(subtask_domain)
-        timesheets_by_subtask = defaultdict(lambda: request.env['account.analytic.line'].sudo())
-        for timesheet in subtasks_timesheets:
-            timesheets_by_subtask[timesheet.task_id] |= timesheet
+
         values['allow_timesheets'] = task.allow_timesheets
         values['timesheets'] = timesheets
```

Comment:
And so I'm not even sure that those 2 keys are needed

Reply:
> And so I'm not even sure that those 2 keys are needed

`allow_timesheets` is still used in project_task_portal_template and `timesheets` in both view.